webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/pdf/GeneralScrambleSheet.kt

```diff
@@ -80,43 +74,20 @@ class GeneralScrambleSheet(scrambleSet: ScrambleSet, activityCode: ActivityCode)
         }
     }
 
-    private fun getFontConfiguration(scrambleColumnWidth: Float, scrambleImageSize: Float, scrambles: List<String>): Font {
-        val availableScrambleHeight = scrambleImageSize - 2 * SCRAMBLE_IMAGE_PADDING
-
-        val availableArea = Rectangle(scrambleColumnWidth - 2 * SCRAMBLE_PADDING_HORIZONTAL,
-            (availableScrambleHeight - SCRAMBLE_PADDING_VERTICAL_TOP - SCRAMBLE_PADDING_VERTICAL_BOTTOM))
-
-        val longestScramble = scrambles.maxBy { it.length } ?: ""
-
-        val longestAlignedScramble = scrambles.map { StringUtil.padTurnsUniformly(it, WIDEST_CHAR_STRING) }
-            .maxBy { it.length } ?: longestScramble
+    private fun getFontConfiguration(availableArea: Rectangle, scrambles: List<String>): Font {
+        val longestScramble = scrambles.flatMap { it.split(NEW_LINE) }.maxBy { it.length } ?: ""
+        val maxLines = scrambles.map { it.split(NEW_LINE) }.map { it.count() }.max() ?: 1
 
-        val longestScrambleOneLine = "\n" !in longestAlignedScramble//Masked
+        val fontSize = PdfUtil.fitText(Font(FontUtil.MONO_FONT), longestScramble, availableArea, FontUtil.MAX_SCRAMBLE_FONT_SIZE, true) // FIXME const
+        val fontSizeIfIncludingNewlines = availableArea.height / maxLines
 
-        // I don't know how to configure ColumnText.fitText's word wrapping characters,
-        // so instead, I just replace each character I don't want to wrap with M, which
-        // should be the widest character (we're using a monospaced font,
-        // so that doesn't really matter), and won't get wrapped.
-        val longestScrambleMasked = longestScramble.replace(".".toRegex(), WIDEST_CHAR_STRING)
-        val longestAlignedScrambleMasked = longestAlignedScramble.replace("\\S".toRegex(), WIDEST_CHAR_STRING)
-        val longestAlignedScrambleMaskedAndStuffed = longestAlignedScrambleMasked.replace(" ".toRegex(), WIDEST_CHAR_STRING)
-
-        val fontSizeForMaskedUnaligned = PdfUtil.fitText(Font(FontUtil.MONO_FONT), longestScrambleMasked, availableArea, FontUtil.MAX_SCRAMBLE_FONT_SIZE, false, 1f) // FIXME const
-
-        // If the scramble contains newlines, then we *only* allow wrapping at the newlines.
-        val longestRespectingNewlines = longestAlignedScrambleMasked.takeIf { longestScrambleOneLine }
-            ?: longestAlignedScrambleMaskedAndStuffed
-        val fontSizeIfIncludingNewlines = PdfUtil.fitText(Font(FontUtil.MONO_FONT), longestRespectingNewlines, availableArea, FontUtil.MAX_SCRAMBLE_FONT_SIZE, true, 1f) // FIXME const
-
-        val oneLine = longestScrambleOneLine && fontSizeForMaskedUnaligned >= FontUtil.MINIMUM_ONE_LINE_FONT_SIZE
-        val perfectFontSize = fontSizeForMaskedUnaligned.takeIf { oneLine } ?: fontSizeIfIncludingNewlines
+        val perfectFontSize = min(fontSize, fontSizeIfIncludingNewlines)
```

Comment:
You are basically discarding the distincion between one-line scrambles and line-wrapped scrambles. Can you add w few words on why you think this is okay?

Reply:
This is what allowed removing megaminx hack. I'll add comments on the code.

Reply:
Also, this removed the annoying M hack. There were a lot of strange things going on like `longestAlignedScrambleMaskedAndStuffed`